### PR TITLE
Bump Jedi upper pin to <0.20

### LINKED ISSUE #414: Support Jedi 0.19
- This was released a couple of weeks ago and comes with support for Python 3.11 and other improvements.
- Creating an issue about it so I don't forget to address it for 1.8

### PR DESCRIPTION
**Pull request by bnavigator**
There doesn't seem to be any breaking change in jedi 0.19

Closes #414

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ readme = "README.md"
 license = {text = "MIT"}
 requires-python = ">=3.7"
 dependencies = [
-    "jedi>=0.17.2,<0.19.0",
+    "jedi>=0.17.2,<0.20.0",
```
Comment (ccordoba12) [inline pyproject.toml:16]:
```suggestion
    "jedi>=0.18.0,<0.20.0",
```

0.17 is quite old now and probably doesn't support the latest Python versions.

**Review by ccordoba12 — APPROVED** [review]
Thanks @bnavigator for your help with this!

File: pyproject.toml (hunk 1 above)
Comment (bnavigator) [reply pyproject.toml:16]: Au contraire. If you don't need >=0.18 in your library, don't change it. Let the user environment and the dependencies some room and don't artificially constrain stuff, you don't need to. I know we had similar conversations in the past, and I respect your differing opinion, but I don't see any need to change this here.

Comment (ccordoba12) [reply pyproject.toml:16]: Ok, no problem with that approach either.

**Pull request merged**